Add <main> to "Offset post without featured image"

carolinan opened the pull request:
<!-- Thanks for contributing to Twenty Twenty-Five! Please follow the Contributing Guidelines:
https://github.com/WordPress/twentytwentyfive/blob/trunk/README.md#contributing -->

**Description**
Adds the `<main>` HTML tag to the pattern. All templates must include a main, both because all content needs to be inside a landmark, but also because in block themes, the automated skip link requires it.

Indent the blocks in the file to improve the readability of the code.

Partial for https://github.com/WordPress/twentytwentyfive/issues/278

**Screenshots**



**Testing Instructions**

Apply the PR and go to Appearance > Editor > Templates.
Select the single post template.
In the settings sidebar, open the Template tab.
Open the Design panel and select the "Offset post without featured image" design.
Open the list view.
Select the group block that is between the header and footer.
In the block settings sidebar, open the Advanced panel.
Confirm that main is selected in the HTML element option.
Save.
View a single post on the front.
Confirm that there are no issues with the layout, the template should look identical before and after adding the main.

juanfra commented on patterns/offset-post-no-featured-image-template.php:
```diff
@@ -14,92 +14,119 @@
-<!-- /wp:group --></div>
+<!-- wp:group {"tagName":"main","align":"wide","layout":{"type":"default"}} -->
+<main class="wp-block-group alignwide">
+	<!-- wp:group {"className":"is-style-default","style":{"spacing":{"padding":{"top":"var:preset|spacing|80","bottom":"var:preset|spacing|40"},"margin":{"top":"0","bottom":"0"}}},"layout":{"type":"constrained"}} -->
```
Let's remove the `is-style-default`

juanfra commented on patterns/offset-post-no-featured-image-template.php:
```diff
@@ -14,92 +14,119 @@
+<!-- wp:group {"tagName":"main","align":"wide","layout":{"type":"default"}} -->
+<main class="wp-block-group alignwide">
+	<!-- wp:group {"className":"is-style-default","style":{"spacing":{"padding":{"top":"var:preset|spacing|80","bottom":"var:preset|spacing|40"},"margin":{"top":"0","bottom":"0"}}},"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group is-style-default" style="margin-top:0;margin-bottom:0;padding-top:var(--wp--preset--spacing--80);padding-bottom:var(--wp--preset--spacing--40)">
```
Let's remove the `is-style-default`

juanfra commented on patterns/offset-post-no-featured-image-template.php:
```diff
@@ -14,92 +14,119 @@
+	</div>
+	<!-- /wp:group -->
+
+	<!-- wp:group {"className":"is-style-default","style":{"spacing":{"padding":{"top":"var:preset|spacing|30","bottom":"var:preset|spacing|50"},"margin":{"top":"0","bottom":"0"}}},"layout":{"type":"constrained"}} -->
```
Let's remove the `is-style-default`

juanfra commented on patterns/offset-post-no-featured-image-template.php:
```diff
@@ -14,92 +14,119 @@
+	<!-- /wp:group -->
+
+	<!-- wp:group {"className":"is-style-default","style":{"spacing":{"padding":{"top":"var:preset|spacing|30","bottom":"var:preset|spacing|50"},"margin":{"top":"0","bottom":"0"}}},"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group is-style-default" style="margin-top:0;margin-bottom:0;padding-top:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--50)">
```
Let's remove the `is-style-default`

juanfra reviewed (commented):
Looks good - I found a couple of `is-style-default` appearances that we may want to remove now.

carolinan commented:
It looks like there were many changes to the file, but that is mostly because of the indentation.

I removed `is-style-default`.
I renamed it to template-single-offset based on the discussion here: https://github.com/WordPress/twentytwentyfive/issues/244
And I replaced the comments area markup with the comments pattern, to avoid repeating code.
I checked that the previous comment area was identical to the pattern.

To do:

- [ ] Internationalization - add translation functions
- [ ] Check if the Post navigation pattern can be used, to avoid repeating code.

carolinan commented:
I found that the spacing is different between the post navigation in this pattern and in the the post navigation pattern.
It can be sorted, but yeah lets get these fixes in first.
